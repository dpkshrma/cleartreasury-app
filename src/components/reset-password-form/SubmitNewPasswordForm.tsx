import { Button, Input } from "@clear-treasury/design-system";
import React, { FunctionComponent, Ref } from "react";

export interface SubmitNewPasswordFormProps {
  onSubmit: () => void;
  loading: boolean;
  invalidCode: boolean;
  passwordRef: Ref<HTMLInputElement | null>;
}

const SubmitNewPasswordForm: FunctionComponent<SubmitNewPasswordFormProps> = ({
  loading,
  passwordRef,
  invalidCode,
  onSubmit,
}) => {
  return (
    <React.Fragment>
      <div className="mb-6">
        <Input
          name="newPassword"
          type="password"
          label="Password"
          placeholder="Password 8+ characters"
          ref={passwordRef}
        />
      </div>
      <Button
        loading={loading}
        size={Button.Size.LARGE}
        onClick={onSubmit}
        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
        // @ts-ignore
        disabled={invalidCode || loading}
      >
        Submit
      </Button>
    </React.Fragment>
  );
};

export default SubmitNewPasswordForm;
