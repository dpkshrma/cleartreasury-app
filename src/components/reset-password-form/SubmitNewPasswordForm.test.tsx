import * as React from "react";
import { render } from "../../../test/testUtils";
import * as Stories from "./SubmitNewPasswordForm.stories";

describe("SubmitNewPasswordForm", () => {
  it("matches snapshot for default state", () => {
    const { asFragment } = render(
      <Stories.Default {...Stories.Default.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
