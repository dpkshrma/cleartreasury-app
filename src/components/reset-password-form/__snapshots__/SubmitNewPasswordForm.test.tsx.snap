// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitNewPasswordForm matches snapshot for default state 1`] = `
<DocumentFragment>
  <div
    class="mb-6"
  >
    <div
      class="w-full relative "
    >
      <label
        class="text-base text-gray-700 inline-block mb-2"
        for="newPassword"
        id="newPassword-label"
      >
        Password
      </label>
      <input
        class="form-input border border-gray-300 bg-white mb-2 p-3 w-full focus:ring-teal-100 focus:border-teal-300 disabled:cursor-not-allowed placeholder-gray-400 rounded"
        id="newPassword"
        name="newPassword"
        placeholder="Password 8+ characters"
        type="password"
        value=""
      />
    </div>
  </div>
  <button
    class="inline-flex items-center justify-center gap-2
    border border-transparent
    ring-theme-button-primary
    text-center leading-snug
    rounded-theme-radius
    focus:outline-none focus:ring-1 focus:ring-offset-1 focus:ring-offset-transparent
    transition-all duration-300
    hover:shadow-md hover:cursor-pointer bg-theme-button-primary text-theme-color-on-primary hover:bg-theme-button-primary-hover py-3 px-8"
  >
    Submit
  </button>
</DocumentFragment>
`;
